src/models/user.ts

```diff
@@ -0,0 +1,97 @@
+// Import the necessary modules
+import mongoose, { Document, Schema } from 'mongoose';
+
+interface Address {
+  city: string;
+  district: string;
+  neighborhood: string;
+  streetAddress: string;
+}
+
+// Define the user schema
+interface UserSchema extends Document {
+  firstname: string;
+  lastname: string;
+  email: string;
+  phone: string;
+  profileImage: string;
+  password: string;
+  role: 'admin' | 'cook' | 'customer';
+  isActive: boolean;
+  address: string;
+}
+
+const addressSchema: Schema<Address> = new Schema({
+  city: {
+    type: String,
+    required: [true, 'City is required']
+  },
+  district: {
+    type: String,
+    required: [true, 'District is required']
+  },
+  neighborhood: {
+    type: String,
+    required: [true, 'Neighborhood is required']
+  },
+  streetAddress: {
+    type: String,
+    required: [true, 'Street address is required']
+  }
+});
```

Comment:
no need to create a new schema for the address since the address will be just a nested object in the user model.

Reply:
I agree with you. No need to create new schema for address.